sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -212,3 +183,25 @@ private[sql] case class UserDefinedAggregator[IN, BUF, OUT](
 
   override def name: String = givenName.getOrElse(aggregator.name)
 }
+
+private[sql] object UserDefinedFunctionUtils {
```

Comment:
Should we name it `UserDefinedFunction` as a companion object?

Reply:
No, UserDefinedFunction will be implementation agnostic. This should be moved to internal as well.

Reply:
Moved it.